File: add/codesnippet/CPP/6af8e8fe-c7a9-45c1-9170-_1.cpp
      String^ myUrl = "http://localhost/Sample_cs.vsdisco";
      DiscoveryClientProtocol^ myProtocol = gcnew DiscoveryClientProtocol;
      DiscoveryDocumentReference^ myReference = gcnew DiscoveryDocumentReference( myUrl );
      Stream^ myFileStream = myProtocol->Download( myUrl );
      DiscoveryDocument^ myDiscoveryDocument = dynamic_cast<DiscoveryDocument^>(myReference->ReadDocument( myFileStream ));

File: add/codesnippet/CSharp/6af8e8fe-c7a9-45c1-9170-_1.cs
         string myUrl = "http://localhost/Sample_cs.vsdisco";
         DiscoveryClientProtocol myProtocol = new DiscoveryClientProtocol();
         DiscoveryDocumentReference myReference = new DiscoveryDocumentReference(myUrl);
         Stream myFileStream = myProtocol.Download(ref myUrl);
         DiscoveryDocument myDiscoveryDocument = 
                           (DiscoveryDocument)myReference.ReadDocument(myFileStream);